motor/discovery/azure/compute.go

```diff
@@ -14,44 +14,29 @@ import (
 	azure_transport "go.mondoo.com/cnquery/motor/providers/azure"
 )
 
-type AzureClient struct {
-	Subscription string
-	Authorizer   autorest.Authorizer
-}
-
-func (c *AzureClient) VirtualMachinesClient() compute.VirtualMachinesClient {
+func (c *Compute) VirtualMachinesClient() compute.VirtualMachinesClient {
```

Comment:
nit: can we move the method definitions underneath the definition of `type Compute struct`. This would improve readability.